### ISSUE TITLE
Make the challenges translatable

### ISSUE THREAD

**Issue by overflorian** [osmlab/maproulette3#1679]
Hi, following [the translation of the whole service in french](https://github.com/osmlab/maproulette3/issues/1675), we now suggest that the challenge/projects themselves could be translated.
Indeed, there is very few projects in french, so for a beginer it means that most of the challenges (including the vital instructions) are not filtered nor translated, leading to a poor UX.
See below the missing translation for this example https://maproulette.org/challenge/21827/task/105985939
![image](https://user-images.githubusercontent.com/11553211/146217381-125b14ed-62ef-4e06-b432-0a6f3e6f78f5.png)
![image](https://user-images.githubusercontent.com/11553211/146217705-b481b900-95f0-45d3-afb5-eaac58d8aa1d.png)

Therefore we suggest you add the wording contributed while creating a challenge to Transifex so we can translate them too.

**Comment by overflorian** [comment]
After discussing it with @panieravide I believe that it could be a solution that the translation is managed directly while creating a new challenge in the website's interface.
It could be managed directly by any user (even without an account) the same way it's managed in the OSM wiki, see
![image](https://user-images.githubusercontent.com/11553211/146645126-50c0a2ed-314d-4db6-b30b-3bbf98f11f31.png)

Another solution proposed by @gendy54 would be to have an automated translation for that (Google translate or equivalent (why not "Libre translate"). Or, even better, a automated translation as a suggestion that would be asked to confirm by the user. Maybe a dedicated button would be useful for that.

Least but not last, it could be that we keep not to manage this translation but that the french challenges are displayed first for a french user. It would require the creator of a new challenge pick the language in the process.

What would be your best pick?

**Comment by gendy54** [comment]
I was also thinking of the WeeklyOSM translation system: https://osmbc.openstreetmap.de/login

**Comment by mvexel** [comment]
Hi @overflorian @gendy54 . I agree this is a 'gap' in the localization of MapRoulette. We have not implemented challenge-level localization under the assumption that a challenge is either targeted at a national / single language audience and thus would not require this, or that challenge is for a global audience and English would be the appropriate language, as is the case for most OSM documentation and tooling. 

Introducing localization at the challenge level, short of providing a deepl / google translate link for convenience, would require a significant re-architecting of the challenge backend to store the different languages, as well as the challenge maintenance front end. This is something we can consider for a next major version of MapRoulette if there is enough support for it. So I am going to leave this open, but without significant code contribution I don't see this happening before the next major version. 

In the interim, is something like this https://github.com/vsetka/deepl-translator an option to add dynamic on-demand translation of component text @jschwarz2030 ?

**Comment by matheusgomesms** [comment]
I agree with Martijn on this, but just a quick thought: not sure if this is easier than implementing something like Transifex or automatic translations by DeepL, but what if the challenge admin had the option to translate those fields in the administration area, just like Tasking Manager does?

**Comment by mvexel** [comment]
I am going to leave this open because I can clearly see the value in this, but with the caveat that it is a significant amount of work, so I don't want to raise expectations that this is going to be implemented anytime soon. 

If someone with React skills wants to help work on it, I would welcome any offer to help out with this feature.

**Comment by mvexel** [comment]
https://twitter.com/maproulette/status/1506382711660765184 :D
